### PR TITLE
Testing several pages of Android SDK code snippets

### PR DESCRIPTION
**Pull request by nathan-contino-mongo**
### Docs staging link (requires sign-in on MongoDB Corp SSO):
Access Custom User Data: https://docs-mongodbcom-staging.corp.mongodb.com/realm/docsworker-xlarge/even-moar-android-tests/android/access-custom-user-data
Create & Manage User API Keys: https://docs-mongodbcom-staging.corp.mongodb.com/realm/docsworker-xlarge/even-moar-android-tests/android/create-manage-api-keys
Manage Email/Password Users: https://docs-mongodbcom-staging.corp.mongodb.com/realm/docsworker-xlarge/even-moar-android-tests/android/manage-email-password-users

### CODE REVIEW

File: source/android/create-manage-api-keys.txt [hunk 1]
```diff
@@ -48,103 +48,66 @@ create user API keys for anonymous users.
...
 
 .. _android-api-key-look-up-user-key:
 
-Look up a User API Key
-----------------------
+Look up a User's API Keys
+-------------------------
 
 To get a list of all user API keys associated with the logged in user,
 call the ``fetchApiKeys()`` or ``fetchApiKeysAsync()`` methods of a
```
Comment (nlarew) [inline source/android/create-manage-api-keys.txt:68]: Can we:
- update these method names (and others throughout) to the correct names
- link method names to the API reference the first time they come up in an example (like in the [newly ported Node examples](https://docs-mongodbcom-staging.corp.mongodb.com/realm/docsworker-xlarge/new-ia/sdk/node/examples/manage-user-api-keys#look-up-a-user-api-key))

e.g. `` :java-sdk:`ApiKeyAuth.fetchAll() <io/realm/mongodb/auth/ApiKeyAuth.html#fetchAll-->` ``

**Review by nlarew — CHANGES_REQUESTED** [review]
LGTM on the tests/includes. Left a comment requesting a general review pass / updates for method names that are incorrect/unlinked.

File: source/android/create-manage-api-keys.txt (hunk 1 above)
Comment (nathan-contino-mongo) [reply source/android/create-manage-api-keys.txt]: Done.

**Comment by nathan-contino-mongo** [comment]
@nlarew please take a look when you get a chance, addressed your comment!

File: source/android/create-manage-api-keys.txt [hunk 2]
```diff
@@ -48,202 +50,136 @@ create user API keys for anonymous users.
...
 
-Look up a User API Key
-----------------------
+Look up a User's API Keys
+-------------------------
 
 To get a list of all user API keys associated with the logged in user,
-call the ``fetchApiKeys()`` or ``fetchApiKeysAsync()`` methods of a
-logged in user's ``apiKeyAuth`` instance.
+call the synchronous :java-sdk:`fetchll() <io/realm/mongodb/auth/ApiKeyAuth.html#fetchAll-->`
```
Comment (nlarew) [inline source/android/create-manage-api-keys.txt:70]:
```suggestion
call the synchronous :java-sdk:`fetchAll() <io/realm/mongodb/auth/ApiKeyAuth.html#fetchAll-->`
```

^ Typo fix. Tangentially, it's kinda weird that the sync/async methods here overload `fetchAll()` but earlier there were separate `create()` and `createAsync()` methods.

File: source/android/create-manage-api-keys.txt [hunk 3]
```diff
@@ -48,202 +50,136 @@ create user API keys for anonymous users.
...
+method or asynchronous :java-sdk:`fetchAll()
+<io/realm/mongodb/auth/ApiKeyAuth.html#fetchAll-io.realm.mongodb.App.Callback->` method of a
+logged in user's :java-sdk:`apiKeyAuth <io/realm/mongodb/auth/ApiKeyAuth.html>` instance.
```
Comment (nlarew) [inline source/android/create-manage-api-keys.txt:73]:
```suggestion
logged in user's :java-sdk:`ApiKeyAuth <io/realm/mongodb/auth/ApiKeyAuth.html>` instance.
```

nit: I think for all the occurrences of this we should either use the capitalized class name (i.e. `ApiKeyAuth`) or the User property that contains the instance (i.e. `User.apiKeys`). Thoughts?

File: source/android/manage-email-password-users.txt [hunk 4]
```diff
@@ -26,77 +26,50 @@ Register a New User Account
...
 .. _android-email-password-confirm-user:
 
 Confirm a New User's Email Address
 ----------------------------------
 
 To confirm a newly-created user, pass a confirmation ``token`` and
-``tokenId`` to the ``confirmUser()`` or ``confirmUserAsync`` methods of
-your {+service-short+} ``App``'s ``EmailPasswordAuth`` instance:
+``tokenId`` to the :java-sdk:`confirmUser() <io/realm/mongodb/auth/EmailPasswordAuth.html#confirmUser-java.lang.String-java.lang.String->`
+or :java-sdk:`confirmUserAsync <io/realm/mongodb/auth/EmailPasswordAuth.html#confirmUserAsync-java.lang.String-java.lang.String-io.realm.mongodb.App.Callback->`
```
Comment (nlarew) [inline source/android/manage-email-password-users.txt:56]:
```suggestion
or :java-sdk:`confirmUserAsync() <io/realm/mongodb/auth/EmailPasswordAuth.html#confirmUserAsync-java.lang.String-java.lang.String-io.realm.mongodb.App.Callback->`
```

nit: method name should include parentheses

File: source/android/manage-email-password-users.txt [hunk 5]
```diff
@@ -115,69 +88,40 @@ To reset a user's password, first send the user a password reset email:
...
-
-         app.emailPassword.sendResetPasswordEmailAsync(emailAddress) {
-            if (it.isSuccess) {
-                Log.i(TAG, "Successfully sent the user a reset password link to $emailAddress")
-            } else {
-                Log.e(TAG, "Failed to send the user a reset password link to $emailAddress")
-            }
-         }
-   
+       :tabid: kotlin
```
Comment (nlarew) [inline source/android/manage-email-password-users.txt:91]: GitHub won't let me comment on the intro text for this example, but can we link to reference docs for `sendResetPasswordEmailAsync()`?

Comment (nlarew) [reply source/android/manage-email-password-users.txt:91]: Also out of scope for this ticket but notably this section doesn't have any info on function-based password resets, i.e. `callResetPasswordFunction()`/`callResetPasswordFunctionAsync()`

**Review by nlarew — APPROVED** [review]
Some small tweaks for API reference links but otherwise LGTM

File: source/android/create-manage-api-keys.txt (hunk 3 above)
Comment (nathan-contino-mongo) [reply source/android/create-manage-api-keys.txt]: done

**Pull request merged**